frontend/apps/app/app/api/chat/route.ts

```diff
@@ -121,8 +123,11 @@ export async function POST(request: Request) {
     `chat-${validationResult.output.designSessionId}-${payloadHash}`,
   )
 
-  // Trigger the Deep Modeling workflow with idempotency key
-  await deepModelingWorkflowTask.trigger(jobPayload, {
+  // Trigger the appropriate workflow based on Deep Modeling toggle
+  const task = validationResult.output.isDeepModelingEnabled
+    ? deepModelingWorkflowTask
+    : designProcessWorkflowTask
```

Comment:
Since the design process alone is not expected to exceed the Vercel timeout of 900 seconds, it would be faster to respond in route.ts, but in this case, the processes are temporarily combined.